### PR TITLE
Tests for pre-transaction-actions plugin, improve post-transaction-actions plugin test

### PR DESCRIPTION
**Pull request by jrohel**
Created from tests for post-transaction-actions.

Requires https://github.com/rpm-software-management/dnf-plugins-core/pull/523

### CODE REVIEW

**Comment by jrohel** [comment]
It seems that CI il is not related to this PR.

**Comment by kontura** [comment]
> It seems that CI il is not related to this PR.

Yes that is an issue caused by libsolv update, here is a fix to update our CI: https://github.com/rpm-software-management/ci-dnf-stack/pull/1452

File: dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    """
+
+Examples:
+    | state     | output            |
+    | any       | install setup     |
+    | in        | install setup     |
+    | out       |                   |
+
+
+Scenario: Do not traceback when reason change is in transaction
```
Comment (kontura) [inline dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature:84]: The `Scenario` name mentions reason change but I don't see any changing reasons in the test.

File: dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  Given I create and substitute file "/etc/dnf/plugins/pre-transaction-actions.d/test.action" with
+    """
+    *:any:echo '${{state}} ${{name}}-${{epoch}}:${{ver}}-${{rel}}.${{arch}} repo ${{repoid}}' > {context.dnf.installroot}/trans.log
+    """
+    And I use repository "installonly"
+    And I configure dnf with
+        | key                          | value         |
+        | installonlypkgs              | installonlyA  |
+        | installonly_limit            | 2             |
+   When I execute dnf with args "-v install installonlyA-1.0"
```
Comment (kontura) [inline dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature:94]: Is the `-v` flag required in these executions?

File: dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature (hunk 1 above)
Comment (jrohel) [reply dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature:84]: @kontura 
The behaviour of the `pre-transaction-actions` plugin is the same as that of `post-transaction-actions` plugin. So I reused (copied) tests for `post-transaction-actions` plugin.
The author of this test is @jmracek . I will check the tests.

File: dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature (hunk 2 above)
Comment (jrohel) [reply dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature:94]: No. I will remove it.

File: dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        | Action        | Package                         |
+        | install       | installonlyA-2.0-1.x86_64       |
+    And stderr does not contain "Traceback"
+    And file "/trans.log" contents is
+    """
+    install installonlyA-0:2.0-1.x86_64 repo installonly
+    """
+   When I execute dnf with args "-v install installonlyA-2.2"
+   Then the exit code is 0
+    And Transaction is following
```
Comment (j-mracek) [inline dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature:116]: Reason change happen when install and remove kernel in one transaction. This is a workaround that fix logic of history DB in background of DNF.

**Comment by kontura** [comment]
Thank you, the failing tests are not related.

File: dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature (hunk 3 above)
Comment (j-mracek) [reply dnf-behave-tests/dnf/plugins-core/pre-transaction-actions.feature:124]: It looks like that reason change is added only in DNF5, and not in DNF4

**Pull request merged**